s3bucket/main.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_s3_bucket" "codedeploy_bucket" {
+  bucket = "${var.name_prefix}-codedeploy-releases"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_iam_policy" "codedeploy_policy" {
```

Comment:
Why? An instance profile and a policy are two completely different things. With this approach you can attach the policy to either a role (created with the instance profile module), a user, a group, or even all together. I don't see why it makes it hard to reuse it with the instance profile module.